sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/action/ActionsImpl.scala

```diff
@@ -83,10 +85,26 @@ final class ActionService(
 private[javasdk] object ActionsImpl {
   private[action] val log = LoggerFactory.getLogger(classOf[ActionsImpl])
 
-  private def handleUnexpectedException(service: ActionService, command: ActionCommand, ex: Throwable): ActionResponse =
+  private def handleUnexpectedException(
+      service: ActionService,
+      command: ActionCommand,
+      ex: Throwable): ActionResponse = {
+    ex match {
+      case badReqEx: BadRequestException => handleBadRequest(service, command, badReqEx)
```

Comment:
We'd need the same kind of handling for all components, because all components can have path/query param annotations, and need to turn this exception into a failure response with INVALID_ARGUMENT.

That's probably why you say the error code does not reach the user, the test is for an entity, not because the proxy is missing logic.

Reply:
It is already handled for all entities, but the proxy ignores the status code for Failures different than the ones from the effect builder.

Reply:
for instance here: https://github.com/lightbend/kalix-jvm-sdk/pull/1619/files#diff-9a0d8d88f0ecd39710e5c4242e4115bb9300efeed6112df3949db5859805d1aeR116

Reply:
I don't get it, we have test coverage here https://github.com/lightbend/kalix-proxy/blob/main/proxy/core/src/test/scala/kalix/proxy/action/ActionServiceSpec.scala#L140

For the exception we send a failure response with a custom code and that should reach the client. Or are you saying that doesn't work?

Reply:
It works for actions, but not for entities: https://github.com/lightbend/kalix-proxy/blob/main/proxy/core/src/main/scala/kalix/proxy/eventsourcedentity/EventSourcedEntity.scala#L381

Reply:
Aha, so what is missing is error code propagation for entities, but it's at least the same protocol so it's just about picking the `grpc_status_code` up from the `Failure` message. 

Does that mean we don't have the custom error codes effect API in the entities either (probably, if we had it we would have noticed it didn't work)

Reply:
ok, so we have two possible sources of failures:
unexpected failures, on the very top level of the protocol:
```
message EventSourcedStreamOut {
  oneof message {
    EventSourcedReply reply = 1;
    Failure failure = 2;
    EventSourcedSnapshotReply snapshot_reply = 3;
  }
}
```
user-defined failures:
```
message EventSourcedReply {
  // The id of the command being replied to. Must match the input command.
  int64 command_id = 1;
  // The action to take
  ClientAction client_action = 2;
  ...
  }

message ClientAction {
  oneof action {
    // Send a reply
    Reply reply = 1;
    // Forward to another component
    Forward forward = 2;
    // Send a failure to the client
    Failure failure = 3;
  }
}
```

for the second one, you can propagate the error code as expected. The question is which kind of a failure I should use for request validation errors?

Reply:
I think the client action level of failure is fine here, a bad request doesn't break state or affect subsequent requests to the same entity, it's just something we want to communicate to the client, just like an error effect from the user.

Reply:
Ok, VE, ES, WE, Action done. I've put the catch for entities in a different place than for the action, feels more suitable to check it there, but I am open to suggestions.

Reply:
Looks good to me